bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,425 @@
+                    break;
+            }
+        }
+
+    }
+
+    @NonNullByDefault
+    private class EventMonitor extends EventCallback {
+
+        @Override
+        public boolean acceptEvent(int event_code) {
+            return true;
+        }
+
+        @SuppressWarnings({ "null" })
+        @Override
+        public void processEvent(@Nullable CGateSession cgate_session, int eventCode,
+                @Nullable GregorianCalendar event_time, @Nullable String event) {
+            if (event == null)
+                return;
+            LinkedList<String> tokenizer = new LinkedList<String>(Arrays.asList(event.trim().split("\\s+")));
+
+            if (eventCode == 701) {
+                @Nullable
+                String address = tokenizer.poll();
+                tokenizer.poll();
+                @Nullable
+                String value = tokenizer.poll();
+                if (value != null && value.startsWith("level=")) {
+                    String level = value.replace("level=", "");
+                    updateGroup(address, level);
+                }
+            }
+        }
+
+    }
+
+    private void updateGroup(@Nullable String address, @Nullable String value) {
+        if (address == null || value == null)
+            return;
+        String[] addressParts = address.trim().replace("//", "").split("/");
+        int application = Integer.parseInt(addressParts[2]);
+        int group = Integer.parseInt(addressParts[3]);
+        updateGroup(addressParts[1], application, group, value);
+    }
+
+    private void updateGroup(String network, int application, int group, String value) {
+        for (Thing networkThing : getThing().getThings()) {
+            // Is this networkThing from the network we are looking for...
+            if (networkThing.getThingTypeUID().equals(CBusBindingConstants.BRIDGE_TYPE_NETWORK) && networkThing
+                    .getConfiguration().get(CBusBindingConstants.CONFIG_NETWORK_ID).toString().equals(network)) {
+                ThingHandler netThingHandler = networkThing.getHandler();
+                if (netThingHandler == null) {
+                    continue;
+                }
+                // Loop through all the things on this network and see if they match the application / group
+                for (Thing thing : ((CBusNetworkHandler) netThingHandler).getThing().getThings()) {
+                    ThingHandler thingThingHandler = thing.getHandler();
+                    if (thingThingHandler == null) {
+                        continue;
+                    }
+
+                    if (thingThingHandler instanceof CBusGroupHandler) {
+                        ((CBusGroupHandler) thingThingHandler).updateGroup(application, group, value);
+                    }
+                }
+            }
+        }
+    }
+
+    @Nullable
+    public CGateSession getCGateSession() {
+        return cGateSession;
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
```

Comment:
super.dispose() should be called at the end of the method.

Reply:
Fixed in both places i found this.